textrepo-app/src/main/java/nl/knaw/huc/TextRepoApp.java

```diff
@@ -147,7 +148,15 @@ public void run(TextRepoConfiguration config, Environment environment) {
    */
   private ViewBuilderFactory createViewBuilderFactory() {
     var viewBuilderFactory = new ViewBuilderFactory();
+
     viewBuilderFactory.register("text", TextViewerResource::new);
+
+    // XmlViewer returns a JSON list of XML snippets. No contents helper is used to compress, e.g. a whole file.
+    // If more views get added here that also don't need contents helper, or perhaps need different parameters,
+    // we should refactor this to get rid of the "throw contentsHelper away" kludge used here.
+    // Let's add first build up some more experience by implementing more viewers, then tackle this.
```

Comment:
Wat is de vervolgstap? En wanneer gaan we die zetten?

Reply:
JSON viewer nog en Hennie's viewer. Dan hebben we er 4 en gaan we kijken wat de gemeenschappelijke deler is en hoe we wat daarbuiten valt dan wel door kunnen geven.